scripts/gas_diff_stats.py

```diff
@@ -142,12 +151,19 @@ def stat(old, new):
         legacy_optimized = stat(parsed[1], parsed[4])
```

Comment:
It was not changed by this PR, but shouldn't we assert parsed there, before `ir_optimized` assignment?
```python
        assert(len(parsed) == 6)
```

Reply:
Yeah, good idea. With less we'll fail anyway because we access all those items but with more we'd continue without failing.

But overall there's a lot more that could be improved. For example `parse_git_diff()` would be better off returning a dict so that it's harder to mix up the indices. I intentionally tried not to get into refactors like that though :) This is just the minimum needed for the change I wanted.

The assert is just a single though so I added it now.